cirq/sim/clifford/clifford_simulator.py

```diff
@@ -366,14 +379,26 @@ def apply_single_qubit_unitary(self, op: 'cirq.Operation'):
         phase_shift = u[max_idx] / applied_unitary[max_idx]
         self.ch_form.omega *= phase_shift
 
-    def _apply_H(self, qubit: ops.Qid):
+    def _apply_H(self, qubit: int):
         self.tableau._H(qubit)
         self.ch_form._H(qubit)
 
-    def _apply_S(self, qubit: ops.Qid):
+    def _apply_S(self, qubit: int):
         self.tableau._S(qubit)
         self.ch_form._S(qubit)
 
+    def _apply_X(self, qubit: int):
+        self.tableau._X(qubit)
+        self.ch_form._X(qubit)
+
+    def _apply_Z(self, qubit: int):
+        self.tableau._Z(qubit)
+        self.ch_form._Z(qubit)
+
+    def _apply_Y(self, qubit: int):
+        self.tableau._Y(qubit)
+        self.ch_form._Y(qubit)
```

Comment:
I think we've decided that Clifford gates will define their effect on the tableau/CH form. IOW, this should be defined in the respective gates, not special cased here. Instead, here you should be invoking a protocol. The protocol would check whether a gate defines an effect on tableau/CH form and if so would use it for efficiency. If not, it would fall back to computing the effect.

This is also consistent with `_apply_unitary_`.

@Strilanc @fedimser 

(Also, I wonder whether it makes sense to keep both the tableau and CH form.)

Reply:
Thanks for the reminder. This is stated in https://github.com/quantumlib/Cirq/issues/2423. Also, with respect to your last statement; here is my take on it: https://github.com/quantumlib/Cirq/issues/2423#issuecomment-549546013

Reply:
`_clifford_tableau_` hasn't been created yet so we can block this and wait or merge and have it be fixed once `_clifford_tableau_`  has been created. I'm down for either as this speeds up the simulation.

Reply:
Given that using the current implementation (which was recently added) is significantly slowing down the simulation, I'd vote for letting this go through. 

I will work on implementing the `_clifford_tableau_` method and protocol and clean this up ASAP.

Reply:
@viathor are you okay with letting this optimization merge in and I can clean up later.